Wox/Storage/TopMostRecord.cs

```diff
@@ -1,47 +1,52 @@
 ﻿using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.InteropServices;
+using Newtonsoft.Json;
 using Wox.Infrastructure.Storage;
 using Wox.Plugin;
 
 namespace Wox.Storage
 {
+    // todo this class is not thread safe.... but used from multiple threads.
     public class TopMostRecord
     {
-        public Dictionary<string, Record> records = new Dictionary<string, Record>();
+        [JsonProperty]
+        private Dictionary<string, Record> records = new Dictionary<string, Record>();
 
         internal bool IsTopMost(Result result)
         {
+            if (records.Count == 0)
+            {
+                return false;
+            }
+
+            // since this dictionary should be very small (or empty) going over it should be pretty fast. 
             return records.Any(o => o.Value.Title == result.Title
-                && o.Value.SubTitle == result.SubTitle
-                && o.Value.PluginID == result.PluginID
-                && o.Key == result.OriginQuery.RawQuery);
+                                     && o.Value.SubTitle == result.SubTitle
+                                     && o.Value.PluginID == result.PluginID
+                                     && o.Key == result.OriginQuery.RawQuery);
```

Comment:
Hey @theClueless can I draw your attention to this issue here(https://github.com/jjw24/Wox/issues/84) from @HsunGong which suggests that top records should match if the query contains the keyword, as seen here it does an exact compare at the moment. What do you think, would this enhance the experience?

Reply:
I honestly don't like this feature anyway :) but I think making it use contains can create some odd issues, also note that it also compare against the original raw query so as long as you do the same "exc" (for excel) it will bring it on top.